map/search_api.cpp

```diff
@@ -222,6 +223,34 @@ bool SearchAPI::SearchInViewport(ViewportSearchParams const & params)
   return Search(p, false /* forceSearch */);
 }
 
+bool SearchAPI::SearchForDiscovery(DiscoverySearchParams const & params)
+{
+  CHECK(params.m_onResults, ());
+  CHECK(!params.m_query.empty(), ());
+  CHECK_GREATER(params.m_itemsCount, 0, ());
+
+  SearchParams p;
+  p.m_query = params.m_query;
+  p.m_inputLocale = "en";
+  p.m_viewport = params.m_viewport;
+  p.m_position = params.m_viewportCenter;
+  p.m_maxNumResults = params.m_itemsCount;
+  p.m_mode = search::Mode::Viewport;
+  p.m_onResults = [params](Results const & results) {
+    if (!results.IsEndMarker())
+      return;
+
+    if (params.m_sortingType == DiscoverySearchParams::SortingType::None)
+      params.m_onResults(results);
+
+    Results r(results);
+    sort(r.begin(), r.end(), DiscoverySearchParams::HotelsSort());
```

Comment:
Здесь их надо перенумеровать с помощью ```SetPositionInResults```.

Reply:
ok

Reply:
@mpimenov, can we get rid of this position-in-results? As was discussed, it would be nice to split results by types: suggests, results, lat-lons. After that there will be no need in "position in results".

Reply:
Sure, but it is a matter for another PR.